### ISSUE TITLE
Preserve SELinux file context on saving

### ISSUE THREAD

**Issue by andrey-utkin** [neovim/neovim#26127]
### Problem

This happens on SELinux-enabled Fedora 39 system.
I edit /etc/crontab with neovim.
After I edit it, crond fails to read it:

```
Nov 20 22:48:01 homeserver crond[2538]: (*system*) RELOAD (/etc/crontab)
Nov 20 22:48:01 homeserver crond[2538]: ((null)) Unauthorized SELinux context=system_u:system_r:system_cronjob_t:s0-s0:c0.c1023 file_context=unconfined_u:obje>
Nov 20 22:48:01 homeserver crond[2538]: (root) FAILED (loading cron table)
```
The workaround is to restore that file's expected context with `restorecon -Fv /etc/crontab`.

Before editing, the file has this context:
```
 # ls -lZ /etc/crontab
-rw-r--r--. 1 root root system_u:object_r:system_cron_spool_t:s0 938 Nov 20 22:51 /etc/crontab
```
After:
```
 # ls -lZ /etc/crontab
-rw-r--r--. 1 root root unconfined_u:object_r:etc_t:s0 938 Nov 20 23:10 /etc/crontab
```

I guess it's because the file is edited and saved not in place, but using a copy which is then "moved into" the old name.

This is a common reoccurring problem: people apparently edit system cron configs and later find they stop working:
https://bugzilla.redhat.com/show_bug.cgi?id=1625645
https://bugzilla.redhat.com/show_bug.cgi?id=1547368

We can't immediately fix this for all editors, and I am not even completely sure the editor is the best party to take responsibility. But I believe NeoVIM can set the precedent for others to follow.

### Expected behavior

SELinux context remains the same after saving.

**Comment by justinmk** [comment]
Is this related to ACL support? https://github.com/neovim/neovim/issues/1200

**Comment by andrey-utkin** [comment]
Yes. It doesn't seem like there is ongoing work, or a lead. Whom shall I contact to try to contribute anything? Or shall i just join dev irc chatroom?

**Comment by justinmk** [comment]
Yeah, we are definitely interested in ACL support, but we need someone to own it. Joining the dev chat is a good idea, but https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md should also answer most questions.
